core/src/rust/filodb_core/src/state.rs

```diff
@@ -0,0 +1,144 @@
+//! State objects shared with Java
```

Comment:
The git diff is a bit confused with this file.  It is mostly a move as is, with the execute_cachable_query methods being new.